File: convalida-validators/src/main/java/convalida/validators/PasswordValidator.java
package convalida.validators;

import android.widget.EditText;
import java.util.regex.Pattern;

/**
 * @author  Wellington Costa on 21/06/2017.
 */
public class PasswordValidator extends AbstractValidator {

    private int min;
    private String pattern;

    public PasswordValidator(
            EditText editText,
            int min,
            String pattern,
            String errorMessage,
            boolean autoDismiss) {
        super(editText, errorMessage, autoDismiss);
        this.min = min;
        this.pattern = pattern;
    }

    @Override public boolean isNotValid(String value) {
        boolean hasError = value.isEmpty();

        if (min > 0) {
            hasError |= value.length() < min;
        }

        if (!pattern.isEmpty()) {
            hasError |= !Pattern.matches(pattern, value);
        }

        return hasError;
    }

}

File: convalida-validators/src/main/java/convalida/validators/PasswordValidator.kt
package convalida.validators

import android.widget.EditText
import java.util.regex.Pattern

/**
 * @author  Wellington Costa on 21/06/2017.
 */
class PasswordValidator(editText: EditText, private var min: Int, private val pattern: String, errorMessage: String) : AbstractValidator(editText, errorMessage) {

    override fun isNotValid(value: String): Boolean {
        var hasError = value.isEmpty()

        if (min > 0) {
            hasError = hasError or (value.length < min)
        }

        if (!pattern.isEmpty()) {
            hasError = hasError or !Pattern.matches(pattern, value)
        }

        return hasError
    }

}
